supervision/annotators/core.py

```diff
@@ -1763,3 +1763,113 @@ def validate_custom_values(
+        self.color: Union[Color, ColorPalette] = color
+        self.color_lookup: ColorLookup = color_lookup
+        self.position = position
+        icon = cv2.imread(icon_path, cv2.IMREAD_UNCHANGED)
+        if icon is None:
+            print(f"Error: Couldn't load the icon image from {icon_path}")
+            return
+
+        resized_icon_h, resized_icon_w = (
+            int(icon.shape[0] * icon_size),
+            int(icon.shape[1] * icon_size),
+        )
+        self.icon = cv2.resize(
+            icon, (resized_icon_h, resized_icon_w), interpolation=cv2.INTER_AREA
+        )
+
+    def annotate(
+        self,
+        scene: np.ndarray,
+        detections: Detections,
+        custom_color_lookup: Optional[np.ndarray] = None,
+    ) -> np.ndarray:
+        """
+        Annotates the given scene with icons based on the provided detections.
+        Args:
+            scene (np.ndarray): The image where bounding boxes will be drawn.
+            detections (Detections): Object detections to annotate.
+            custom_color_lookup (Optional[np.ndarray]): Custom color lookup array.
+                Allows to override the default color mapping strategy.
+        Returns:
+            np.ndarray: The annotated image.
+        Example:
+            ```python
+            >>> import supervision as sv
+            >>> image = ...
+            >>> detections = sv.Detections(...)
+            >>> icon_annotator = sv.IconAnnotator(icon_path='path_of_icon')
+            >>> annotated_frame = icon_annotator.annotate(
+            ...     scene=image.copy(),
+            ...     detections=detections
+            ... )
+            ```
+        """
+        icon_h, icon_w = self.icon.shape[:2]
+
+        xy = detections.get_anchors_coordinates(anchor=self.position)
+        for detection_idx in range(len(detections)):
+            color = resolve_color(
+                color=self.color,
+                detections=detections,
+                detection_idx=detection_idx,
+                color_lookup=(
+                    self.color_lookup
+                    if custom_color_lookup is None
+                    else custom_color_lookup
+                ),
+            )
+            # Convert Color type to numpy list in BGRA format, and color icon with it
+            colored_icon = np.ones_like(self.icon) * (list(color.as_bgr()) + [1])
+
+            # The current positions of the anchors don't account for annotations that
+            # have mass. This visually centers the anchor, given the size of the icon
+            x_offset = 0
+            if self.position in [
+                Position.TOP_CENTER,
+                Position.CENTER,
+                Position.BOTTOM_CENTER,
+            ]:
+                x_offset = icon_w / 2
+
+            x, y = (
+                int(xy[detection_idx, 0] - x_offset),
+                int(xy[detection_idx, 1]),
+            )
+
+            # Takes the alpha channel from the original icon
```

Comment:
I think we can remove most of those comments and replace them with more descriptive variable names.